Sync/SyncStateMachine.swift

```diff
@@ -73,10 +73,20 @@ open class SyncStateMachine {
     var stateLabelsSeen = [SyncStateLabel: Bool]()
     var stateLabelSequence = [SyncStateLabel]()
 
+    let stateLabelsAllowed: Set<SyncStateLabel>
+
     let scratchpadPrefs: Prefs
 
-    public init(prefs: Prefs) {
+    /// Use this set of states to constrain the state machine to making a minimal 
+    /// number of HTTP requests. This is suitable for extension uses.
```

Comment:
I'd limit mention of "HTTP requests" — the things we're really concerned about are time and battery, so that's all kinds of things: total HTTP requests, duration, records processed, database writes, fsyncs…